DB/Item.lua

```diff
@@ -165,4 +165,83 @@ function Item:HasZone(entry)
 	return false
 end
 
+function Item:IsCollectionItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.COLLECTION
+end
+
+function Item:IsUsingCollectionProperties(entry)
+	return entry.collectedItemId ~= nil
+end
+
+function Item:IsArchaeologyItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.ARCH
+end
+
+function Item:IsUsingArchaeologyProperties(entry)
+	return entry.raceId ~= nil
+end
+
+function Item:HasUseMethod(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.USE
+end
+
+function Item:HasUseMethodProperties(entry)
+	return entry.items ~= nil
+end
+
+function Item:IsCovenantSpecificItem(entry)
+	return entry.requiresCovenant ~= nil
+end
+
+function Item:IsUsingCovenantProperties(entry)
+	return entry.requiredCovenantID ~= nil
```

Comment:
Perhaps this could be narrowed down to actually enforce the field's content to be one of the recognized constants?

Reply:
Checking for actual valid covenantIds in e1e77bdd39b5bbc878999175be753afbe2f9af44